### ISSUE TITLE
Support multiple deploy environments

### ISSUE THREAD

**Issue by Janpot** [mui/mui-toolpad#403]
Make applications parametrizable and allow setting up multiple parameter/secrets groups to deploy a release to multiple environments.
Each environment has its independent parameter group and its independent version.

**Comment by oliviertassinari** [comment]
I would personally add the `waiting for upvotes` label to wait. My assumption would be that this feature would target people that are not yet completely willing to make the speed vs. maintainability tradeoff of low-code. They might be the audience the hardest to convert to Toolpad, so why I wouldn't start there.

**Comment by Janpot** [comment]
Yes, for multiple environments I agree. I do think there is some value to split application definition from configuration early on. It could be helpful in implementing arbitrary secrets, and also in solving some of the issues we talked about around having secrets readable in the editor or not
